internal/linker/test/link_node_modules.spec.ts

```diff
@@ -32,6 +32,48 @@ describe('link_node_modules', () => {
     mkdirp(workspace);
   });
 
+  it('should group and reduce modules appropriate', () => {
+    const modules = {
+      'a': ['src', `${workspace}/path/to/lib_a`],
+      '@foo/c/c/c/c': ['bin', `${workspace}/path/to/foo_cccc`],
+      'b/b': ['src', 'other_wksp/path/to/lib_bb'],
+      'b': ['src', 'other_wksp/path/to/lib_b'],
+      '@foo/c': ['bin', `${workspace}/path/to/foo_c`],
+      '@foo/c/c': ['bin', `${workspace}/path/to/foo_cc`],
+      '@foo/d/bar/fum/far': ['bin', `${workspace}/path/to/foo_d/bar/fum/far`],
+      '@foo/d/bar': ['bin', `${workspace}/path/to/foo_d/bar`],
+      // don't include `@foo/d` as the group & reduce function  should derive that
+      // from the lowest common denominator of the module name & module path
+    };
+
+    const result = linker.groupAndReduceModules(modules);
+
+    const expected = [
+      [
+        // `@foo/c`, `@foo/c/c` and `@foo/c/c/c/c` should
+        // be grouped but not reduced as their arrangement
+        // on disk does not match the modules names
+        {'name': '@foo/c', 'root': 'bin', 'modulePath': `${workspace}/path/to/foo_c`},
+        {'name': '@foo/c/c', 'root': 'bin', 'modulePath': `${workspace}/path/to/foo_cc`},
+        {'name': '@foo/c/c/c/c', 'root': 'bin', 'modulePath': `${workspace}/path/to/foo_cccc`}
+      ],
+      [
+        // `@foo/d/bar/fum/far` and `@foo/d/bar` should
+        // be grouped and reduced to `@foo/d`
+        {'name': '@foo/d', 'root': 'bin', 'modulePath': `${workspace}/path/to/foo_d`},
+      ],
+      [{'name': 'a', 'root': 'src', 'modulePath': `${workspace}/path/to/lib_a`}],
+      [
+        // `b`, `b/b` should be grouped but not reduced as
+        // their arrangement on disk does not match the modules names
```

Comment:
do you know whether this case occurs today? it feels like maybe we should error for this until we know why it should be supported. could someone be relying on this behavior outside of Bazel?

Reply:
That's a good question. Outside of bazel someone could do something like this in theory. @jbedard may be relying on this with the current rollup_bundle.

Reply:
Discussed in person. Decided that it is better to allow this behavior as it is possible to do something similar outside of bazel with module mapping.